sdk/python/foundation-models/system/inference/text-generation/llama-files/score/hf-tgi/score.py

```diff
@@ -0,0 +1,460 @@
+            }, 
+            { 
+                "role": "user", 
+                "content": "and in Europe?" 
+            } 
+        ], 
+        "parameters":{ 
+            "max_length": 100,
+            "temperature": 0.9,
+            "top_p": 0.6,
+            "do_sample": true,
+            "max_new_tokens":100 
+        } 
+    } 
+}
+
+"""
+
+def get_processed_input_data_for_cc(data: List[str]) -> str:
+    """
+    example input:
+    [
+        {"role": "user", "content": "What is the tallest building in the world?"},
+        {"role": "assistant", "content": "As of 2021, the Burj Khalifa in Dubai"},
+        {"role": "user", "content": "and in Africa?"},
+    ]
+    example output:
+    "[INST]What is the tallest building in the world?[\INST]As of 2021, the Burj Khalifa in Dubai\n[INST]and in Africa?[/INST]"
+    """
+    B_INST, E_INST = "[INST]", "[/INST]"
+    conv_arr = data
+    history = ""
+    assert len(conv_arr) > 0
+    assert conv_arr[0]["role"] == "user"
+    history += B_INST + conv_arr[0]["content"].strip() + E_INST
+    assert conv_arr[-1]["role"] == "user"
+    for i, conv in enumerate(conv_arr[1:]):
+        if i % 2 == 0:
+            assert conv["role"] == "assistant"
+            history += conv["content"].strip() + "\n"
+        else:
+            assert conv["role"] == "user"
+            history += B_INST + conv["content"].strip() + E_INST
+    return history
+
+
+def get_request_data(request_string) -> Tuple[Union[str, List[str]], Dict[str, Any]]:
+    """
+    return type for cc: str, dict
+    return type for tg: list, dict
+    """
+    global task_type
+    try:
+        data = json.loads(request_string)
+        logger.info(f"data: {data}")
+        # hf-tgi expects "inputs", while mir inference payloads expect "input_data"
+        inputs = data.get("input_data", data.get("inputs", None))
+
+        input_data = []   # type: Union[str, List[str]]
+        params = {} # type: Dict[str, Any]
+        if isinstance(inputs, dict):
+            input_data = inputs["input_string"]
+            params = inputs.get("parameters", {})
+        elif isinstance(inputs, str):
+            input_data = inputs
+            params = data.get("parameters", {})
+        else:
+            raise Exception("input_data is not a dict or string")
+
+        if not isinstance(params, dict):
+            raise Exception("parameters is not a dict")
+
+        if task_type == SupportedTask.CHAT_COMPLETION:
+            if not isinstance(input_data, list):
+                raise Exception("input_str is not a list (for cc)")
+            print("CC task. Processing input data")
```

Comment:
nit: I feel it's better to use chat completion instead of cc in comments and logs, as this code is customer facing.